crates/rattler_repodata_gateway/src/gateway/remote_subdir.rs

```diff
@@ -48,9 +48,13 @@ impl RemoteSubdirClient {
             e => GatewayError::FetchRepoDataError(e),
         })?;
 
+        let repo_data_json_path = repodata.repo_data_json_path.clone();
+        // repodata holds onto a file lock for json file that sparse will need
+        drop(repodata);
+
         // Create a new sparse repodata client that can be used to read records from the repodata.
         let sparse = LocalSubdirClient::from_channel_subdir(
-            &repodata.repo_data_json_path,
+            &repo_data_json_path,
```

Comment:
I was reacquiring because `fetch_repo_data` acquires a `rw` lock while I was hoping to use a `ro` lock in `SparseRepoData` so there could be multiple processes reading the same file without blocking.  

When testing on windows yesterday I stopped getting deadlocks after dropping the lock in the python bindings but after having the test time out again I setup an env on a linux vm and I'm still getting deadlocks.  

After reexamining this change, I don't think this will work.  The scenario I'm thinking of is lets say there's two files and one process holds the ro lock for one while waiting for the rw lock for the other while the 2nd process holds the ro lock that the 1st process is waiting for the rw lock on while also waiting on the rw lock for the other file.

Reply:
Two possible alternatives I'm thinking of are
1. Introducing a barrier so that a process finishes all `fetch_repo_data` calls and drops all the locks at the same time before trying to acquire locks for the `SparseRepoData`
2. Adding option to favor `from_bytes` instead of the memory mapped option in `SparseRepoData`